src/main/java/com/aws/greengrass/mqtt/bridge/clients/LocalMqtt5Client.java

```diff
@@ -0,0 +1,419 @@
+            ConnAckPacket packet = onConnectionFailureReturn.getConnAckPacket();
+            LogEventBuilder l = LOGGER.atError().kv("error", CRT.awsErrorString(errorCode));
+            if (packet != null) {
+                l.kv("reasonCode", packet.getReasonCode().name())
+                        .kv(LOG_KEY_REASON, packet.getReasonString());
+            }
+            l.log("Failed to connect to Local Mqtt5 Client");
+        }
+
+        @Override
+        @SuppressWarnings("PMD.DoNotLogWithoutLogging")
+        public void onDisconnection(Mqtt5Client client, OnDisconnectionReturn onDisconnectionReturn) {
+            int errorCode = onDisconnectionReturn.getErrorCode();
+            DisconnectPacket packet = onDisconnectionReturn.getDisconnectPacket();
+            // Error code 0 means that the disconnection was intentional. We do not need to run callbacks when we
+            // purposely interrupt a connection.
+            if (errorCode == 0 || packet != null && packet.getReasonCode()
+                    .equals(DisconnectPacket.DisconnectReasonCode.NORMAL_DISCONNECTION)) {
+                LOGGER.atInfo().log("Connection purposefully interrupted");
+            } else {
+                LogEventBuilder l = LOGGER.atWarn().kv("error", CRT.awsErrorString(errorCode));
+                if (packet != null) {
+                    l.kv("reasonCode", packet.getReasonCode().name())
+                            .kv(LOG_KEY_REASON, packet.getReasonString());
+                }
+                l.log("Connection interrupted");
+            }
+
+            synchronized (updateSubscriptionsLock) {
+                if (updateSubscriptionsTask != null) {
+                    updateSubscriptionsTask.cancel(true);
+                }
+            }
+        }
+
+        @Override
+        public void onStopped(Mqtt5Client client, OnStoppedReturn onStoppedReturn) {
+            client.close();
+        }
+    };
+
+    /**
+     * Construct a LocalMqtt5Client.
+     *
+     * @param brokerUri          broker uri
+     * @param clientId           client id
+     * @param mqttClientKeyStore KeyStore for MQTT Client
+     * @param executorService    Executor service
+     * @throws MessageClientException if unable to create client for the mqtt broker
+     */
+    public LocalMqtt5Client(@NonNull URI brokerUri,
+                            @NonNull String clientId,
+                            MQTTClientKeyStore mqttClientKeyStore,
+                            ExecutorService executorService) throws MessageClientException {
+        this.brokerUri = brokerUri;
+        this.clientId = clientId;
+        this.mqttClientKeyStore = mqttClientKeyStore;
+        this.mqttClientKeyStore.listenToCAUpdates(this::reset);
+        this.executorService = executorService;
+
+        // TODO configure ssl
+        Mqtt5ClientOptions mqtt5ClientOptions =
+                new Mqtt5ClientOptions.Mqtt5ClientOptionsBuilder(brokerUri.getHost(), (long) brokerUri.getPort())
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

`getPort()` returns -1 if the port is not set. Check the value returned by `getPort`. Use a default port if the value returned is negative. See the following diffs on GitHub that make this code change: [diff1](https://github.com/apache/skywalking/commit/10d55860dbb4a4c6fec20d54c1a61382bc681481#diff-d4c3e5510bbb6120dc300aa7b2b1c9051d39fd70662af287db112833c7944e11L43), [diff2](https://github.com/opennetworkinglab/onos/commit/6381a24e9d48d4d4e7110cf78b91a6ab47f791d0#diff-7678808b29b9776d35b4ec9d00428db2fa2e2adcfc6ffc7a69812f3ab330fe10L114)

Learn more: [URL.getPort in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/net/URL.html#getPort--)

Reply:
https://github.com/aws-greengrass/aws-greengrass-mqtt-bridge/pull/107